db/blip_handler.go

```diff
@@ -1128,22 +1128,18 @@ func (bh *blipHandler) processRev(rq *blip.Message, stats *processRevStats) (err
 		var currentBucketDoc *Document
 
 		// Look at attachments with revpos > the last common ancestor's
-		minRevpos := 1
 		if len(history) > 0 {
 			currentDoc, rawDoc, err := bh.collection.GetDocumentWithRaw(bh.loggingCtx, docID, DocUnmarshalSync)
 			// If we're able to obtain current doc data then we should use the common ancestor generation++ for min revpos
 			// as we will already have any attachments on the common ancestor so don't need to ask for them.
 			// Otherwise we'll have to go as far back as we can in the doc history and choose the last entry in there.
 			if err == nil {
-				commonAncestor := currentDoc.History.findAncestorFromSet(currentDoc.CurrentRev, history)
-				minRevpos, _ = ParseRevID(bh.loggingCtx, commonAncestor)
-				minRevpos++
 				rawBucketDoc = rawDoc
 				currentBucketDoc = currentDoc
-			} else {
-				minRevpos, _ = ParseRevID(bh.loggingCtx, history[len(history)-1])
 			}
 		}
+		// updatedRevPos is the revpos of the new revision, to be added to attachment metadata if needed for CBL<4.0 compatibility. revpos is not used
```

Comment:
Can we identify which versions of CBL require revpos, if any?  I was under the impression that revpos is strictly a SGW concept and shouldn't actually be required for any clients as long as SGW is using 1:1 attachment:document storage.

Reply:
I don't think any code requires revpos, I looked back to release/2.8. Some lite code looks like it sends revpos to the server but I do not think this matters.

I ran the tests in couchbase-lite-core that use real sync gateway but I think https://github.com/couchbaselabs/couchbase-lite-tests/blob/4d49b6f9f71744ce8b121b74922f35ae717e3d76/tests/test_replication_blob.py#L95 does an absolute diff on the document and that's why I didn't remove populating revpos in https://github.com/couchbase/sync_gateway/commit/8986872dddf91054811fccaed7bc78e094c8cc88

revpos is also used by couchdb, but I'm not sure how.